APKBUILD.5

```diff
@@ -0,0 +1,488 @@
+Specifies the run-time dependencies of the -dev subpackage.
+
+.It Cm giturl
+Specifies the URL of the Git repository to use with
+.Cm abuild checkout .
+If the default branch of the repository is not desired, a different one may be
+specified by appending
+.Fl b Ar branch
+where
+.Cm branch
+is the branch to checkout.
+.El
+
+
+.Ss Functions
+Functions specified here may be present in any
+.Nm
+file, but with the exception of
+.Li build and
+.Li package ,
+are not strictly required.
+
+.Bl -tag -width Ds
+.It Cm fetch
+This function is called to download the remote files in
+.Cm source .
+
+.It Cm unpack
+This function unpacks any archives in
+.Cm source
+to
+.Ev srcdir .
+
+.It Cm prepare
+Prepares the source in
+.Ev srcdir
+to be built.  The default
+.Cm prepare
+function ensures the build directories are set up correctly and applies any
+*.patch files specified in
+.Cm source .
+You must call
+.Cm default_prepare
+if you write a custom
+.Cm prepare
+function.
+
+.It Cm build
+Compiles the source in
+.Ev builddir .
+You must implement this function yourself.  If no compilation is required, you
+may simply write
+.Li return 0
+to skip the build phase.
+
+.It Cm check
+Runs the package's test suite.  This function must be implemented unless
+.Li Dq !check
+was specified in
+.Cm options .
+
+.It Cm package
+Installs the package into
+.Ev pkgdir .
+Note that
+.Ev pkgdir
+is not created for you; if this package installs no files (for example, a
+metapackage), you must use
+.Li mkdir -p "$pkgdir"
+to skip the package phase.
+
+.El
+
+
+.Ss Install Scripts
+An install script is run when an action is taken on a package by
+.Xr apk 8 .
+An install script must be written in shell and must have a
+.Li Dq #!/bin/sh
+interpreter declaration as the first line.  The
+.Cm install
+variable must contain the install scripts needed by the package.  The different
+actions available are as follows:
+
+.Bl -tag -width Ds
+.It Ic $pkgname.pre-install
+Executed before the package is installed.  If this script exits with an error
+(non-zero exit code),
+.Xr apk 8
+will halt the installation and the package will not be installed.  This install
+script is typically used to create any users or groups needed as described in
+.Cm pkggroups
+and
+.Cm pkgusers .
+
+.It Ic $pkgname.post-install
+Executed after the package is installed.  If this script exits with an error
+(non-zero exit code),
+.Xr apk 8
+will mark the package as broken.  The
+.Li apk fix
+command will attempt to re-run the post-install script if this occurs.
+
+.It Ic $pkgname.pre-upgrade
+Executed before the package is upgraded.  If this script exits with an error
+(non-zero exit code),
+.Xr apk 8
+will mark the package as broken.
+
+.It Ic $pkgname.post-upgrade
+Executed after the package is upgraded.  If this script exits with an error
+(non-zero exit code),
+.Xr apk 8
+will mark the package as broken.  The
+.Li apk fix
+command will attempt to re-run the post-upgrade script if this occurs.
```

Comment:
Document arguments given to the script (new version and old version of the pkg).

Reply:
Okay, will do.

Reply:
Is there any documentation… anywhere… that actually describes the arguments passed to the install scripts?  I didn't see any on the wiki.  I guess I can dive in to apk source code if not.

Reply:
I believe that it is just the version being installed, but don't quote me on that.

Reply:
*-upgrade scripts gets two versions - old and new. I don't remember the order though (I'm on phone now), but you can look e.g. into main/nginx post-upgrade script.

Reply:
(This is why we need this documentation, haha.)

I've added a paragraph about the environment install scripts are run in, and the arguments passed.

Reply:
P.S. I used [the apk-tools source code](https://git.alpinelinux.org/cgit/apk-tools/tree/src/database.c#n2783) as reference.